### PR TITLE
Add negated glob patterns and gitignore support to catalog

### PR DESCRIPTION
**Pull request by jeduden**
## Summary
This PR adds support for negated glob patterns (with `!` prefix) and gitignore integration to the catalog directive, allowing users to exclude files from generated sections.

## Key Changes

- **Negated glob patterns**: Glob patterns prefixed with `!` now exclude matching files from the catalog. For example:
  ```yaml
  glob:
    - "docs/*.md"
    - "!docs/internal.md"
  ```

- **Gitignore support**: Added a new optional `gitignore` parameter (defaults to `true`) that respects `.gitignore` rules when matching files. Can be explicitly set to `"false"` to disable.

- **Validation enhancements**:
  - Validates that `gitignore` parameter is either `"true"` or `"false"`
  - Ensures at least one non-negated (include) pattern exists in the glob list
  - Validates negated patterns for absolute paths, `..` traversal, and syntax errors

- **Implementation details**:
  - Added `splitIncludeExclude()` function to separate include and exclude patterns
  - Added `isExcluded()` function to check if a file matches any exclusion patterns
  - Added `isGitignored()` function to check gitignore rules
  - Integrated gitignore matcher into the file runner
  - Updated `buildCatalogEntries()` to apply both exclusion patterns and gitignore filtering

- **Comprehensive test coverage**: Added 14 new test cases covering:
  - Single and multiple exclusion patterns
  - Wildcard support in exclusions
  - Error cases (absolute paths, `..` traversal, invalid patterns, empty patterns)
  - Edge cases (no matches, only exclusions, with row templates)
  - Gitignore parameter validation and behavior

- **Documentation updates**: Updated example catalogs in PLAN.md and docs/guides/index.md to demonstrate the new exclusion syntax.

https://claude.ai/code/session_01UkjrUXPeunThkPuVDZUjT5

### CODE REVIEW

File: internal/engine/runner.go [hunk 1]
```diff
@@ -58,6 +58,7 @@ func (r *Runner) Run(paths []string) *Result {
 			continue
 		}
 		f.FS = os.DirFS(filepath.Dir(path))
+		f.Gitignore = lint.NewGitignoreMatcher(filepath.Dir(path))
 		if r.RootDir != "" {
```
[the remark this answers is not in the document]
Comment (jeduden) [reply internal/engine/runner.go]: Fixed — `Runner` now caches `GitignoreMatcher` instances per directory via `cachedGitignore()` method, avoiding redundant filesystem walks.

---
_Generated by [Claude Code](https://claude.ai/code)_

[the remark this answers is not in the document]
Comment (jeduden) [reply internal/engine/runner.go]: Fixed — Runner caches matchers per directory via cachedGitignore(). Cache key normalized with filepath.Abs. Uses RootDir when available.

**Comment by jeduden** [comment]
Added documentation in two places:

- **User guide** (`docs/guides/directives/generating-content.md`): new "Excluding files" and "Gitignore filtering" sections with examples
- **Design reference** (`docs/design/archetypes/generated-section/README.md`): "Negated patterns" and "Gitignore filtering" added to list-valued parameters section

See commit 6478fa8.
